lib/install/management/configure.go

```diff
@@ -258,21 +258,25 @@ func (d *Dispatcher) deleteSnapshotByRef(snapshot *types.ManagedObjectReference,
 	return nil
 }
 
-// tryCreateSnapshot try to create upgrade snapshot. It will check if upgrade snapshot already exists. If exists, return error.
-// if succeed, return snapshot refID
-func (d *Dispatcher) tryCreateSnapshot(name, desc string) (*types.ManagedObjectReference, error) {
+// createSnapshot will create a snapshot of the VCH Appliance
+func (d *Dispatcher) createSnapshot(name, desc string) (*types.ManagedObjectReference, error) {
 	defer trace.End(trace.Begin(name, d.op))
 
 	// TODO detect whether another upgrade is in progress & bail if it is.
 	// Use solution from https://github.com/vmware/vic/issues/4069 to do this either as part of 4069 or once it's closed
 
 	info, err := d.appliance.WaitForResult(d.op, func(ctx context.Context) (tasks.Task, error) {
-		return d.appliance.CreateSnapshot(ctx, name, desc, true, false)
+		return d.appliance.CreateSnapshot(ctx, name, desc, false, false)
 	})
 	if err != nil {
 		return nil, errors.Errorf("Failed to create upgrade snapshot %q: %s.", name, err)
 	}
-	return info.Entity, nil
+	snap, ok := info.Result.(types.ManagedObjectReference)
+	if !ok {
+		// TODO: what to do in this situation..
```

Comment:
my 0.02$ here is that we have to fail the attempt to configure/upgrade(I saw this in the configure function, but I am not 100% familiar with the call path). I am surprised we do not attempt the snapshot first in that function. Since we would want the ability to rollback first and foremost. I might be misunderstanding some of the complexity to what is happening in the configure function, but it seems like we should abort if we are not at the very least able to guarantee a return to the old state of the VCH.

Reply:
I might not understand what has happened here as well. Is it possible to get an empty result object on a successful snapshot creation?

Reply:
we do snap before the upgrade...so if the snap fails we will fail the upgrade / configure

Reply:
understood, I was looking at the `Configure` function and I was just noticing that we upload the ISOs(potentially) before we take the snapshot. Not related to this PR, it just seemed out of order.